### DISCUSSION TITLE
dbt unit test support

### DISCUSSION THREAD

**Discussion by s-kethiri** [dbt-athena/dbt-athena#711] [Q&A]
I'm interested in implementing unit tests for my SQL models. dbt Core v1.8+ supports unit testing, but I'm unsure if this is supported specifically for dbt-athena. https://docs.getdbt.com/docs/build/unit-tests

Has anyone got this working?

**Comment by jeancochrane** [comment]
We've been working on adding unit tests to our project, and it seems to be working so far. We ran into a couple of issues we had to fix upstream; https://github.com/dbt-labs/dbt-athena/pull/672 fixed an issue with the way unit tests handle DDL data types, and https://github.com/dbt-labs/dbt-core/pull/10290 / https://github.com/dbt-labs/dbt-adapters/pull/236 fixed the way unit tests (and ephemeral models) construct CTEs using our project's naming scheme. So far so good since merging those three fixes, but I'll post some updates here as our unit tests get more mature.
